contributors/design-proposals/storage/local-storage-pv.md

```diff
@@ -761,3 +761,213 @@ The central manager will poll each worker for their discovered volumes and creat
+
+* `STORAGE_BACKEND` to determine which StorageManager instance to start
+* `PROVISIONER_NAME` to specify the name of the provisioner, its value matches the Provisioner
+specified in the StorageClass.
+
+**Note:** Currently we already have a `provisionerName` in `RuntimeConfig` of the plugin,
+which is in format of `local-volume-provisioner-${NODE_NAME}-${NODE_ID}`, it is annotated
+in PVs with `AnnProvisionedBy`, to indicate which PVs are up to a certain node.
+To distinguish it from the new env `PROVISIONER_NAME`, we will rename it to `provisionerTag`.
+
+A StorageManager instance is responsible for:
+
+1. Monit available capacity for provisioning, and update the capacity in StorageClass if
+there is a change.
+2. Provide the ability to create volumes, detailed in the volume creation section.
+
+A new field `storageSources` will be added into the configuration ConfigMap, to illustrate available
+storage source for provisioning.
+
+```
+kind: ConfigMap
+metadata:
+  name: local-volume-config
+  namespace: kube-system
+data:
+ storageClassMap: |
+    local-static:
+      hostDir: "/mnt/ssds"
+      mountDir: "/local-ssds"
+    local_dynamic:
+      storageSources:
+        - "volumeGroup1"
+      blockCleanerCommand: "/scripts/lv_release.sh"
+```
+For example for an LVM StorageManager, the list of storageSource could contain volume groups.
+For other StorageManagers, the list could contain other storage sources (e.g. unpartitioned disks).
+Each StorageManager should know the scope and what to do with the input.
+
+##### Reconcile Capacity
+
+The StorageManager is expected to reconcile capacity of StorageClasses in its local
+`storageClassMap` regularly. Particularly, LVM StorageManager behaviors as follows:
+
+1. Walks through classes in its `storageClassMap` regularly
+2. Fetch VGs (volume groups) of each class if `storageSources` is specified
+3. Sum up the `VG size`, and compare it with the capacity in StorageClass API object
+4. Update the capacity in StorageClass object if there is a difference
+
+**Note:** For the alpha phase, thinly-provisioned VGs (overcommit) is not supported,
+We'll consider it in a subsequent phase.
+
+Though a list of VGs can be configured as storage source, it's not recommend to
+pass in more than one VG. We may suffer from some edge-cases:
+
+Suppose we have two VGs on a node for provisioning, VG1 and VG2, and they each has 10GB available,
+then the reported capacity would be 20GB, but a pod with storage request of 15GB could not
+be satisfied indeed.
+
+##### Change in Populator
+
+In addition to PV, the `populator` will also be extended to watch StorageClasses and PVC objects of all namespaces.
+
+It stores latest StorageClass objects in cache, they are used:
+
+* To reconcile capacity, the StorageManager need to compare actual capacity with that recorded
+in StorageClass objects
+* During volume creation, some opitions (e.g. fsType for `mkfs`) are specified in StorageClasses

```

Comment:
Can you document in another section what StorageClass parameters the provisioner can support

Reply:
In fact, I haven't thought of other options other than the fsType.